src/main/java/org/folio/service/manager/inputdatamanager/InputDataManagerImpl.java

```diff
@@ -0,0 +1,175 @@
+  @Autowired
+  private Vertx vertx;
+
+  private WorkerExecutor executor;
+  private LocalMap<String, InputDataContext> inputDataLocalMap;
+
+  public InputDataManagerImpl() {
+  }
+
+  public InputDataManagerImpl(Context context) {
+    SpringContextUtil.autowireDependencies(this, context);
+    this.executor = context.owner().createSharedWorkerExecutor(SHARED_WORKER_EXECUTOR_NAME, POOL_SIZE);
+    this.inputDataLocalMap = context.owner().sharedData().getLocalMap(INPUT_DATA_LOCAL_MAP_KEY);
+  }
+
+  @Override
+  public void init(JsonObject request, JsonObject params) {
+    this.executor.executeBlocking(blockingFuture -> initBlocking(request, params), this::handleExportResult);
+  }
+
+  protected void initBlocking(JsonObject request, JsonObject params) {
+    ExportRequest exportRequest = request.mapTo(ExportRequest.class);
+    FileDefinition requestFileDefinition = exportRequest.getFileDefinition();
+    OkapiConnectionParams okapiConnectionParams = new OkapiConnectionParams(params.mapTo(Map.class));
+    FileDefinition fileExportDefinition = createExportFileDefinition(requestFileDefinition);
+    String tenantId = okapiConnectionParams.getTenantId();
+    Iterator<List<String>> sourceStream = sourceReader.getSourceStream(requestFileDefinition, exportRequest.getBatchSize());
+    if (sourceStream.hasNext()) {
+      fileDefinitionService.save(fileExportDefinition, tenantId)
+        .compose(savedFileExportDefinition -> {
+          //TODO
+          //get executionJob and update status to in-progress
+          String jobExecutionId = requestFileDefinition.getJobExecutionId();
+          initInputDataContext(sourceStream, jobExecutionId);
+          ExportPayload exportPayload = createExportPayload(okapiConnectionParams, savedFileExportDefinition, jobExecutionId);
+          exportNextChunk(exportPayload, sourceStream);
+          return Future.succeededFuture();
+        });
+    } else {
+      fileDefinitionService.save(fileExportDefinition.withStatus(FileDefinition.Status.ERROR), tenantId);
+      //TODO
+      //update job execution with error status
+    }
+  }
+
+  @Override
+  public void proceed(JsonObject payload, ExportStatus exportStatus) {
+    ExportPayload exportPayload = payload.mapTo(ExportPayload.class);
+    if (ExportStatus.IN_PROGRESS.equals(exportStatus)) {
+      InputDataContext inputDataContext = inputDataLocalMap.get(exportPayload.getJobExecutionId());
+      Iterator<List<String>> sourceStream = inputDataContext.getSourceStream();
+      if (Objects.nonNull(sourceStream) && sourceStream.hasNext()) {
+        exportNextChunk(exportPayload, sourceStream);
+      } else {
+        finalizeExport(exportPayload, ExportStatus.ERROR);
+      }
+    } else {
+      finalizeExport(exportPayload, exportStatus);
+    }
+  }
+
+  private void exportNextChunk(ExportPayload exportPayload, Iterator<List<String>> sourceStream) {
+    List<String> identifiers = sourceStream.next();
```

Comment:
sourceStream.next() is blocking operation, seems like we need to run proceed method in WorkerExecutor